pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -368,7 +368,18 @@ async def _prepare_request(
 
         # Check usage
         if ctx.deps.usage_limits:  # pragma: no branch
-            ctx.deps.usage_limits.check_before_request(ctx.state.usage)
+            if ctx.deps.usage_limits.count_tokens_before_request:
+                model_request_parameters = await _prepare_request_parameters(ctx)
+                message_history = await _process_message_history(
+                    ctx.state.message_history, ctx.deps.history_processors, build_run_context(ctx)
```

Comment:
This is duplicated with `_make_request`, we can always return these values from this method and use them there instead of building them again